front/app/containers/Admin/reporting/containers/ReportBuilderPage/index.tsx

```diff
@@ -1,125 +1,232 @@
-  Text,
+  colors,
   stylingConsts,
+  Title,
 } from '@citizenlab/cl2-component-library';
 import Tippy from '@tippyjs/react';
+import { useSearchParams } from 'react-router-dom';
 
+import useAuthUser from 'api/me/useAuthUser';
 import useReports from 'api/reports/useReports';
 
 import useFeatureFlag from 'hooks/useFeatureFlag';
 
 import Button from 'components/UI/Button';
+import NavigationTabs from 'components/admin/NavigationTabs';
+import SearchInput from 'components/UI/SearchInput';
+import Tab from 'components/admin/NavigationTabs/Tab';
 
-import { FormattedMessage } from 'utils/cl-intl';
+import { FormattedMessage, useIntl } from 'utils/cl-intl';
+import { isAdmin } from 'utils/permissions/roles';
 
 import CreateReportModal from '../../components/ReportBuilderPage/CreateReportModal';
 import EmptyState from '../../components/ReportBuilderPage/EmptyState';
 import ReportRow from '../../components/ReportBuilderPage/ReportRow';
-import sharedMessages from '../../messages';
+import Warning from 'components/UI/Warning';
 
+import sharedMessages from '../../messages';
 import messages from './messages';
 
+const BuilderNotAllowedTooltip = ({ disabled, children }) => {
+  const { formatMessage } = useIntl();
+
+  return (
+    <Tippy
+      disabled={disabled}
+      placement="bottom"
+      content={formatMessage(sharedMessages.contactToAccess)}
+      hideOnClick
+    >
+      <Box>{children}</Box>
+    </Tippy>
+  );
+};
+
+const TabContainer = ({ children }) => (
+  <Box
+    w="100%"
+    borderBottom={`1px solid ${colors.divider}`}
+    position="relative"
+    zIndex="1"
+    mb="-2px"
+    overflow="hidden"
+  >
+    {children}
+  </Box>
+);
+
+const ListContainer = ({ children }) => (
+  <Box
+    minHeight="66vh"
+    px="36px"
+    pb="36px"
+    border={`1px solid ${colors.divider}`}
+    borderRadius={stylingConsts.borderRadius}
+    bg={colors.white}
+  >
+    {children}
+  </Box>
+);
+
+const tabNames = ['all-reports', 'your-reports', 'service-reports'];
+
 const ReportBuilderPage = () => {
-  const { data: reports } = useReports();
+  const { formatMessage } = useIntl();
+  const [searchParams] = useSearchParams();
+  const { data: me } = useAuthUser();
   const [modalOpen, setModalOpen] = useState(false);
+  const [search, setSearch] = useState<string | undefined>();
+
   const isReportBuilderAllowed = useFeatureFlag({
     name: 'report_builder',
     onlyCheckAllowed: true,
   });
 
-  if (!reports) {
-    return null;
+  // Compute the query parameters for the reports API based on the tab.
+  function getParams(tab: string) {
+    return {
+      search,
+      owner_id: tab === 'your-reports' ? me?.data.id : undefined,
+      service: tab === 'service-reports' ? true : undefined,
+    };
   }
 
+  const { data: yourReports } = useReports(getParams('your-reports'));
+  const { data: serviceReports } = useReports(getParams('service-reports'));
+  const { data: allReports } = useReports(getParams('all-reports'));
```

Comment:
Could consider using the `isLoading` attribute and waiting for those to be `false`, so that you don't get the flashing zeroes before the reports are loaded. E.g.

```
const { data: yourReports, isLoading: isLoadingYourReports } = useReports(getParams('your-reports'));
if (isLoadingYourReports) return null;
```